d2bs/kolbot/libs/modules/Town.js

```diff
@@ -428,6 +448,15 @@
 				}
 			}
 		}
+		else if ( type === "yps" || type === "vps" || type === "wms") {
```

Comment:
Looks to me, as the code within the elseif block, is the same as in the as within if.

```javascript

if (type === "hp" || type === "mp" || type === "yps" || type === "vps" || type === "wms") {
	for (i = 5; i > 0; i -= 1) {
		result = npc.getItem(type + i);
		if (result) {
			return result;
			}
		}
	}
}
```

Would have the same effect and is less verbose

Reply:
The type index is decremented in that. I could put the conditional on the `i` appending, but it's pretty messy there, too. Perhaps a helper function that decides what the item should be.